execution-core/src/transfer.rs

```diff
@@ -93,6 +95,28 @@ impl Serializable<{ StealthAddress::SIZE + u64::SIZE + BlsPublicKey::SIZE }>
     }
 }
 
+/// Data for either contract call or contract deployment.
+#[derive(Debug, Clone, PartialEq, Eq, Archive, Serialize, Deserialize)]
+#[archive_attr(derive(CheckBytes))]
+pub enum CallOrDeploy {
+    /// Data for a contract call.
+    Call(ContractCall),
+    /// Data for a contract deployment.
+    Deploy(ContractDeploy),
+}
```

Comment:
How about calling this `ContractExecution`? or `ContractExec` for short?

Reply:
I'd be inclined to follow Ed's suggestion to use ContractData (like ETH). ContractData has the downside that it is already used in Piecrust though.

Reply:
After some deliberation I think I like ContractExec, it is unique, and in a way, more informative than ContractData.